### DISCUSSION TITLE
Thoughts on current layout & theme of the application

### DISCUSSION THREAD

**Discussion by eliandoran** [TriliumNext/Notes#256] [General]
@TriliumNext/everyone ,

I'll be honest, I'm not very fond of the current theme for Trilium:

![image](https://github.com/user-attachments/assets/9d85d9b9-9b5c-4312-bd0a-95d32d5f34e4)

The theme itself is not the issue actually, but the layout. It feels cluttered. There are so many "dimensions": the button bar on the left, the tree, the tab bar, the header, the ribbon, the sidebar, contextual menus.

To my surprise, I've recently learned that I prefer the old theme:

![image](https://github.com/user-attachments/assets/5f16de24-455c-4a09-84fe-ccf9e0209d87)

In my GNOME theme, I moved the button bar on the top:

![image](https://github.com/user-attachments/assets/24cf4a67-f476-4307-8f3a-dcb7d0eadb88)

In my [Breeze theme](https://github.com/eliandoran/breeze-trilium), I moved the ribbon at the bottom.

![image](https://github.com/user-attachments/assets/66c570e6-098b-429c-8cb6-70051df1d461)

What are your thoughts on improving the default theme? I'll start by suggesting we should probably bring back the sidebar as it was initially.

**Comment by rijulraju** [comment]
Agree wholeheartedly. Here's my custom VS code style theme -- happy to share the source if anyone is interested:
<img width="1440" alt="Screenshot 2024-07-21 at 13 50 56" src="https://github.com/user-attachments/assets/c4707a32-c780-4594-a5d3-03ab5bd9ea80">

**Reply by eliandoran** [reply]
Looks pretty nice. Let me know if there are any design decisions that you think we might want to take into consideration, things that cannot be directly done via themes.

**Reply by CobriMediaJulien** [reply]
@rijulraju i really would like to test your theme. Could you share it with me?

**Comment by eliandoran** [comment]
@Nriver , you mentioned you missed the old logo, did you happen to miss the old theme as well? :)

**Reply by eliandoran** [reply]
If anyone wants a trip down memory lane, I've found a random release containing that theme: https://github.com/zadam/trilium/releases/tag/v0.47.3 

You can run `trilium-portable` and it should work out of the box.

**Reply by Nriver** [reply]
When it comes to button layout, I prefer using the [Lightpad theme](https://github.com/ohmstance/trilium-lightpad-theme).

**On Desktop:**  
I prefer the buttons on the left. I use bookmarks as shortcuts to deep-level notes and usually want to click the note tree immediately after using a bookmark. Having the bookmark and note tree close to each other simplifies navigation. For instance, I have a universal `TODO` note with sub-notes like `daily TODO`, `weekly TODO`, and `monthly TODO`. I bookmark the top `TODO` note and then navigate the tree to find the sub-notes.

**On Mobile:**  
I prefer the buttons on the top. This is more practical due to limited screen space. Buttons on the left would stack on each other due to the mobile IME keyboard which is not easy to touch by fingers. Moving the buttons to the top and making them scrollable solves this problem for me.

**Reply by eliandoran** [reply]
@Nriver , perhaps we could get away with having an adjustable bar that can be placed either at the top or at the left side. Personally , I much prefer the bar at the top, to the left it feels a bit weird.

**Comment by perfectra1n** [comment]
I've tweaked the theme from SiriusXT:
[custom-dark-theme.txt](https://github.com/user-attachments/files/16326275/custom-dark-theme.txt)

I've found it to be a lot easier to see all of my nested notes with it, which was my biggest gripe with the current and previous default theme.

**Reply by eliandoran** [reply]
@perfectra1n , drop a screenshot in here so we can see it without importing. :D

**Reply by doobes** [reply]
I like this theme a lot:

![image](https://github.com/user-attachments/assets/87b4afe2-321a-48f1-a549-be4e0dfd5104)

The lines make it a LOT more readable.

Saying all that, I like the option of having the button bar at the top.

**Reply by meichthys** [reply]
I agree. the tree lines are a must for me. 
I opted for a more subtle version using this css:
```
.fancytree-lastsib ul > li > ul{
    padding-left: 16px;
}
.fancytree-lastsib li{
    box-shadow: inset 1px 0px 0px 0px #333333;
}
```
![image](https://github.com/user-attachments/assets/3ed22ed3-2bec-4989-b9bf-4e33afd99f85)

**Comment by meichthys** [comment]
@eliandoran I really like your GNOME theme 👍
The buttons on the top with the search bar in the top middle feels right to me.

**Reply by meichthys** [reply]
IMO, the QuickSearch and the "Jump to Note" features should be combined into one search/command input. 
I have (mostly new) users that are constantly confused about when to use search and when to use the jump-to feature.

Perhaps adding a `/` (or something else) to the beginning of a search term would return the "jump-to" results instead (similar to how VSCode has the `>` symbol in it's command input field.

**Reply by eliandoran** [reply]
@meichthys , indeed it does seem a bit duplicated IMO.
I actually never use the search bar at the top. I much prefer the Ctrl+J approach.

**Reply by justyns** [reply]
I haven't been keeping up with trilliumnext, but I did have a plugin/script that adds a command palette:

https://github.com/justyns/trilium-scripts/blob/master/command-palette/palette-testing-widget.js

If the beginning of the text input was "> ", it'd show available commands.  Otherwise it'd act like "Jump to Note" and show notes.  I don't think I ever added it, but I planned on having a "Search for ..." always show up as an option in the "Jump to Note" mode. So basically it'd filter a list of notes for what you're typing, but you could also select the search option instead

**Reply by meichthys** [reply]
@eliandoran Likewise, I rarely use the quick search.
@justyns Very interesting. I'll have to give it a try!

**Comment by zerebos** [comment]
I think there's _a lot_ that can be improved UI-wise for Trilium, and of course the current layout and theme is one of them. I would argue finalizing the layout is more important than the theme, but even before doing the layout I think it would be good for us to make the current UI system more proper and more consistent. For example Trilium uses a weird mix of switches and checkboxes amongst other input types all over the place. This is due to it all being jquery with custom html and not using something component-based where you would have a consistent look and feel.

That said, it's never too early to discuss things 😄 as for the launcherbar I both like and dislike it being at the top. Currently I'm for keeping it to the side solely because vertical space is more precious on monitors these days and I prefer not putting something static there that takes away space from the most important part of the note taking app; the note.

I've personally been going back and forth between these two layouts for my custom theme (yet to be made public)

![image](https://github.com/user-attachments/assets/9e11ef46-e91c-4d0d-8b69-38102db20992)
![image](https://github.com/user-attachments/assets/7a2569d9-1404-4b9c-b5cf-35b6d497377d)

The first one is essentially a stylized fancy version of the default layout with some customizations. The second one is more similar to your own theme; the launcherbar is up top. But for mine the tabs are at the bottom and I've extended that to make the tree actions combine with the tabs to make a nice bottom bar. But I like how in your breeze theme you have the note properties at the bottom like that as well.

I also played around with emulating the current Obsidian's look and feel (using a preset customization for my theme) and it seems somewhat suitable to Trilium as well

![image](https://github.com/user-attachments/assets/a6b0a75e-0f65-4cd1-99cd-c45d7a687427)

**Reply by eliandoran** [reply]
@rauenzi ,

> For example Trilium uses a weird mix of switches and checkboxes amongst other input types all over the place. This is due to it all being jquery with custom html and not using something component-based where you would have a consistent look and feel.

Sounds interesting, maybe you could elaborate a bit on this subject and see how we can improve it? Perhaps define which widgets are out of place and which widgets should become template.

>  Currently I'm for keeping it to the side solely because vertical space is more precious on monitors these days and I prefer not putting something static there that takes away space from the most important part of the note taking app; the note.

If vertical space is a concern, then what about the ribbon? An expanded ribbon for "Note info" has 91px (at least on my theme, slight differences might apply), whereas the top bar is under 50px. If I open a daily note, the "Edited Note" ribbon pops at the top **by default** with a whopping 237px of vertical space.

**Reply by zerebos** [reply]
> Sounds interesting, maybe you could elaborate a bit on this subject and see how we can improve it? Perhaps define which widgets are out of place and which widgets should become template.

Just as a quick example of what I mean, in nearly the same spot Trilium will show switch-style booleans and checkbox style

![image](https://github.com/user-attachments/assets/6a461db0-3147-461b-ba58-b0e38f031fef)
vs
![image](https://github.com/user-attachments/assets/5095f9ad-ac5f-4000-908f-cd4270034eac)

It's definitely something I would want to work on. But would you say making the UI consistent while keeping jQuery is important? or should we decide on and switch to a new frontend framework first?

> If vertical space is a concern, then what about the ribbon? An expanded ribbon for "Note info" has 91px (at least on my theme, slight differences might apply), whereas the top bar is under 50px. If I open a daily note, the "Edited Note" ribbon pops at the top by default with a whopping 237px of vertical space.

Oh I agree, I'd love to be able to hide that entire section personally like you can in several other note apps:

![image](https://github.com/user-attachments/assets/fcb8ba41-5af5-45b0-95de-148d952c8fce)
vs
![image](https://github.com/user-attachments/assets/b174a65e-fdd3-4ba4-8a47-9b1c939e3411)

**Reply by eliandoran** [reply]
> It's definitely something I would want to work on. But would you say making the UI consistent while keeping jQuery is important? or should we decide on and switch to a new frontend framework first?

I think it would be a nice idea to do it now. The frontend rewrite might take a while. And if we do it now, I think we'll reduce the workload in the future as well since we'd have to convert all switch-style widgets to a single component instead of having to look for both checkboxes and switch-style widgets.

I think we can create tickets for each of these minor inconsistencies and solve them. But I would do it only after we convert the frontend to TypeScript as well (#156).

> Oh I agree, I'd love to be able to hide that entire section personally like you can in several other note apps:

I thought of that too, but I think the problem is that our ribbon is more than the header of a note, such as what you have in Notion for example. It's a combination of note information, configuration, tags but also bigger elements such as the note map. Some of them have to be relocated to a sidebar I think. Don't forget we already have a sidebar is currently only used for table of contents and highlights list, as far as I know.

**Reply by zerebos** [reply]
> I think it would be a nice idea to do it now. 

I'm of the same mind

> Don't forget we already have a sidebar is currently only used for table of contents and highlights list, as far as I know.

Yeah I think there's a lot of potential for reworking that area of the UI but that might be better to save for after converting the frontend to a modern framework.

**Reply by diyoyo** [reply]
> I've personally been going back and forth between these two layouts for my custom theme (yet to be made public)

Oh my, please make them public :D
I agree with @eliandoran on all the things that are mentioned in the first post.

Personnally, the top things that bug me these days:
- the code coloring. I use a plugin that was suggested a while ago in a message I posted on trilium, but it's really ugly, specially in dark mode. So your (@rauenzi ) theme is making me dream of better days (please don't tell me that the actual code in your screenshot was actually a screenshot image)
- The "Promoted Attributes" : I use a lot of them, as my trilium instance is organized in an Object-Oriented fashion: I have Templates for what I call "Classes" (like Persons, Companies, Movies, Books, ...) and Templates for what I call "Interfaces" (like withDate, withContactInfo, withMoodWeather). So all these generate promoted attributes to children notes. So my dream would be to go away from this "centered random" layout for all the attribute fields. I also would love some kind of field organizer (like in espoCRM).
- The search' result context menu that displays all the attributes of a note, even when the searched keyword is not part of them. 
- the backlink button, which I do like, that hides the text right below it, even when typing.


By the way, these checkboxes are also way too big imho, in the previous screenshot.

Finally, I agree about the inspiration from the other note taking apps. The ones that recently really pleased me in terms of UI are AFFine, and, to some extent, Microsoft Loop, which seems to compile some the recent trends in the note taking scene (including Trilium?)

**Reply by zerebos** [reply]
> Oh my, please make them public :D

I definitely will soon once I fix a few more important issues with it!

> please don't tell me that the actual code in your screenshot was actually a screenshot image

Nope, it's absolutely a code block! It does still use a plugin (likely the same one you're referring to but with a small modification of my own) and my theme overrides the default styles of the plugin.

> The "Promoted Attributes" [...]

Completely agree this system including the display needs a massive overhaul. I agree having some proper field management along with a suitable UI would be game-chaning.

> these checkboxes are also way too big imho, in the previous screenshot.

In my theme I used a bit of a css hack to turn them into switches 
![image](https://github.com/user-attachments/assets/c045e086-5265-4ad2-8072-76ef2f52e857)
![image](https://github.com/user-attachments/assets/4a14c00d-7756-40fc-8a1d-b8afce608e1c)

> The ones that recently really pleased me in terms of UI are AFFine, [...]

AFFine was actually the inspiration for my theme, I tried to capture as much of it as I can including smaller details like the context menus (see below). I have more of a showcase in the Discord since it's not super relevant here 
![image](https://github.com/user-attachments/assets/ca4bfe66-d8e4-47e7-9d4d-932b028c2148)

**Reply by meichthys** [reply]
As opposed to the text-view of attributes,
![image](https://github.com/user-attachments/assets/ba98fdf4-5ccc-41e7-985a-616325ca63d6)

I've always thought it would be nice if the attributes could be formatted a little more like pills (kind-of like how gitea does type-labels):
![image](https://github.com/user-attachments/assets/81eb5323-30a5-4f40-bc34-99bd269da346)

Clicking on the pill could bring up the attribute detail dialog for editing. 

Promoted attributes would be moved to the top of the note view (rather than hidden in the ribbon):
![image](https://github.com/user-attachments/assets/4d5e1929-7b28-4bdc-b65d-f482c45a7f36)


I'd have to think it through a bit more, but i could see us adding an icon to the beginning of the pill to indicate if it is inherited or inheritable, and perhaps coloring specific attribute types similarly.

**Comment by EnBandit** [comment]
https://github.com/JadeVane/Allure
I'm using this theme, and I think it's AWESOME!
Is it possible that the default theme evolve as the theme like this? Or at least there's a built-in alternative like this theme?
A simple theme really makes me more productive, hoping more excellent themes in the future.

**Reply by eliandoran** [reply]
We already have our own theme. You can give it a try in the latest beta version.

**Reply by maphew** [reply]
From the readme, Allure already works with TriliumNext, so I don't understand what the request is for? A streamlined means to install and test themes perhaps. I can see that welcomed by many.

**Comment by abdukir** [comment]
I think a default dark theme would be great. Currently i'm using a little modified version of the https://github.com/cwilliams5/Midnight-Trilium-Dark-Mode (I added the vertical lines in the subtree) Tbh those really help with readability 
![resim](https://github.com/user-attachments/assets/74eec7ec-2a18-449f-8ad0-1b8426c6bf49)

**Reply by eliandoran** [reply]
We are already working on a TriliumNext theme which has dark mode support (both automatic and always dark mode), you can check it out on the latest beta.

**Reply by abdukir** [reply]
Okay will give it a try

**Comment by eliandoran** [comment]
Closing for now as we have already implemented a theme specific for TriliumNext and we also added support for horizontal layout (moving the launch bar at the top).
